First draft of PR template (#4278)
elastic/observability-docs: .github/pull_request_template.md

> ## Description
> <!-- Add a description here -->
> 
> ### Related issue
> Closes # <!-- Add the issue this PR closes here -->
> 
> ## Checklist
> - [ ] Product/Engineering Review
> - [ ] Writer Review
> 
> ### Serverless/Stateful port
> <!--
> Most documentation changes apply to both our Serverless and Stateful content sets.
> You must either add your changes to both content sets or add the `needs-serverless-port`
> or `needs-stateless-port` label to this PR.
> -->
> Select one of the following:
> - [ ] This PR adds content to both the Serverless and Stateful docs ✔️
> - [ ] This PR only applies to one content set and does not need to be ported ✔️
> - [ ] This PR adds Serverless content and has been ported to Stateful in PR#_HERE ✔️
> - [ ] This PR adds Serverless content and I've added the `needs-stateless-port` label 🚧
> - [ ] This PR adds Stateful content and has been ported to Servless in PR#_HERE ✔️
> - [ ] This PR adds Stateful content and I've added the `needs-serverless-port` label 🚧

Author: ```suggestion
## Description
<!-- Add a description here -->

### Documentation sets edited in this PR

_Check all that apply._

- [ ] Stateful (`docs/en/observability/*`)
- [ ] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue
Closes # <!-- Add the issue this PR closes here -->

## Checklist

- [ ] Product/Engineering Review
- [ ] Writer Review

### Follow-up tasks
<!-- If you are updating the Integrations Developer Guide, you can delete this section -->

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [ ] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [ ] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>
```

## Description
<!-- Add a description here -->

### Documentation sets edited in this PR

_Check all that apply._

- [ ] Stateful (`docs/en/observability/*`)
- [ ] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue
Closes # <!-- Add the issue this PR closes here -->

> ## Checklist

Reviewer: Feel free to adjust. I don't know if this should be in a comment or in the text that gets posted to the PR template. On one hand it's easy to dismiss comments... on the other hand it would be repetitive and make PR descriptions so long... 

```suggestion
## Checklist

<!--
Add labels to:
1. Backport to other versions (`backport-*`):
    - `backport-8.x` to backport to the latest minor
    - `backport-skip` to not backport (for example, for serverless docs) 
    - `backport-main` to "backport" to `main` if the target branch is _not_ `main`
    - Individual `backport-*` labels to target specific minor versions
2. Surface blocking reviews (`needs-*-review`):
    - `needs-writer-review` for codeowners
    - `needs-dev-review` for dev team
    - `needs-product-review` for PM review
3. Build serverless preview docs: `ci:doc-build`
-->
```

Author: I think a comment makes sense. Don't want to muddy the PR description too much.

Author: This looks good!

## Description
<!-- Add a description here -->

### Documentation sets edited in this PR

_Check all that apply._

- [ ] Stateful (`docs/en/observability/*`)
- [ ] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue
Closes # <!-- Add the issue this PR closes here -->

## Checklist

<!--
Add labels to:
1. Backport to other versions (`backport-*`):

>     - `backport-8.x` to backport to the latest minor

Reviewer: Last question... do you think we need to be more specific with this? 

```suggestion
    - `backport-8.x` to backport to the latest minor branch (this may be a yet-to-be-released version)
```

Reviewer: >latest minor branch

This might be the wrong phrase because (for example) we don't currently have an 8.16 branch.

Author: I don't think so. We're not doing anything novel in obs-docs so I'm inclined to leave it as-is.
